crates/ra_assists/src/handlers/fill_match_arms.rs

```diff
@@ -56,31 +50,115 @@ pub(crate) fn fill_match_arms(ctx: AssistCtx) -> Option<Assist> {
     }
 
     let db = ctx.db;
-
     ctx.add_assist(AssistId("fill_match_arms"), "Fill match arms", |edit| {
-        let indent_level = IndentLevel::from_node(match_arm_list.syntax());
+        let mut arms: Vec<MatchArm> = match_arm_list.arms().collect();
+        if arms.len() == 1 {
+            if let Some(Pat::PlaceholderPat(..)) = arms[0].pat() {
+                arms.clear();
+            }
+        }
```

Comment:
```suggestion
        if arms.len() == 1 && matches!(arms[0].pat(), Some(Pat::PlaceholderPat(..))) {
            arms.clear();
        }
```
Now with `matches!` if's can be flattened. But idk the general opinion on this new core marco.